spring-security-mvc-boot/src/main/java/org/baeldung/custom/config/SecurityConfig.java

```diff
@@ -1,56 +1,20 @@
 package org.baeldung.custom.config;
 
-import org.baeldung.custom.security.MyUserDetailsService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
-import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.builders.WebSecurity;
-import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @Configuration
-@EnableWebSecurity
-@ComponentScan("org.baeldung.security")
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
 
-    @Autowired
-    private MyUserDetailsService userDetailsService;
-
-    @Override
-    protected void configure(final AuthenticationManagerBuilder auth) throws Exception {
-        auth.authenticationProvider(authenticationProvider());
-    }
-
-    @Override
-    public void configure(WebSecurity web) throws Exception {
-        web.ignoring().antMatchers("/resources/**");
-    }
-
     @Override
-    protected void configure(final HttpSecurity http) throws Exception {
-        // @formatter:off
-        http.authorizeRequests()
-            .antMatchers("/login").permitAll()
-            .antMatchers("/admin").hasRole("ADMIN")
-            .anyRequest().authenticated()
-            .and().formLogin().permitAll()
-            .and().csrf().disable();
-            ;
-        // @formatter:on
-    }
-
-    @Bean
-    public DaoAuthenticationProvider authenticationProvider() {
-        final DaoAuthenticationProvider authProvider = new DaoAuthenticationProvider();
-        authProvider.setUserDetailsService(userDetailsService);
-        authProvider.setPasswordEncoder(encoder());
-        return authProvider;
+    protected void configure(HttpSecurity http) throws Exception {
+        super.configure(http);
+        http.csrf()
```

Comment:
A common mistake is when an application disables this, but still exposes their API over HTTP basic. Since the browser will automatically resend HTTP basic creds, it can't remain a public API and have CSRF disabled.

If we leave it disabled, let's make sure the article briefly calls out why we are justified in doing it in the article (and please remember that readers will be tempted to add this for the convenience).

On the other hand, it's pretty easy to leave this enabled; you can use the `SecurityMockMvcRequestPostProcessors#csrf()` post processor in `MockMvc` to add a test one:

```java
this.mvc.perform(get("/").with(csrf()));
```